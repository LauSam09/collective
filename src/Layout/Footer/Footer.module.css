.footer {
  max-width: var(--page-width);
  margin: 0 auto;
  text-align: right;
}
