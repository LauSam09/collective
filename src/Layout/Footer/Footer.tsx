import classes from "./Footer.module.css"

export default function Footer() {
  const currentYear = new Date().getFullYear()
  return (
    <footer className={classes.footer}>
      <small>&copy; Copyright {currentYear}, Collective</small>
    </footer>
  )
}
