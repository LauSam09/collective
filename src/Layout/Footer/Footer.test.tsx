import { render, screen } from "@testing-library/react"
import renderer from "react-test-renderer"

import { Footer } from "./Footer"

describe("<Footer />", () => {
  test("matches snapshot", () => {
    const json = renderer.create(<Footer />).toJSON()
    expect(json).toMatchSnapshot()
  })

  test("renders contentinfo role", () => {
    render(<Footer />)
    expect(screen.getByRole(/contentinfo/i)).toBeInTheDocument()
  })

  test("renders copyright notice", () => {
    render(<Footer />)
    expect(
      screen.getByText(/Â© Copyright 2021, Collective/i)
    ).toBeInTheDocument()
  })
})
