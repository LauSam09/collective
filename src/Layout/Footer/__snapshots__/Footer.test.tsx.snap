// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches snapshot 1`] = `
<footer
  className="footer"
>
  <small>
    Â© Copyright 
    2021
    , Collective
  </small>
</footer>
`;
